build/miengine.settings.targets

```diff
@@ -1,65 +1,45 @@
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003"> 
   <!--
   Settings file which is included by all the shipping code projects in the repo.
-  -->
-
-  <!-- Validate OS -->
-  <Target Name="OSPlatformCheck" BeforeTargets="Build" Condition="!$([MSBuild]::IsOSPlatform('windows'))" >
-    <Error Text="'MIDebugEngine' does not build non-Desktop on non-windows platforms." Condition="!$(Configuration.Contains('Desktop'))"/>
-  </Target>
-  
+  -->  
   <Import Project="all_projects.settings.targets"/>
   <Import Project="package_versions.settings.targets" />
   <Import Project="version.settings.targets" />
 
   <Choose>
     <When Condition="'$(Lab)' == 'true'">
-        <PropertyGroup>
-            <!--NOTE:DevDivPublicKey.snk comes from src\\tools\\devdiv\\FinalPublicKey.snk -->
-            <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)..\Keys\DevDivPublicKey.snk</AssemblyOriginatorKeyFile>
-
-            <!-- Use PublicSign when TEST_LAB_BUILD is enabled -->
-            <PublicSign Condition="'$(TEST_LAB_BUILD)'!=''">true</PublicSign>
-            <DelaySign Condition="'$(TEST_LAB_BUILD)'==''">true</DelaySign>
-        </PropertyGroup>
-    </When>
-    <Otherwise>
-        <PropertyGroup>
-            <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)..\Keys\ExternalKey.snk</AssemblyOriginatorKeyFile>
-            <DelaySign>false</DelaySign>
-        </PropertyGroup>
-    </Otherwise>
-  </Choose>
-  
-  <Choose>
-    <When Condition="$(Configuration.Contains('Desktop'))">
       <PropertyGroup>
-        <!-- "Desktop.*" configs are used to build the cross-platform bits that are used in the VS Code C++ extension -->
-        <IsXPlat>true</IsXPlat>
-        <DefineConstants>$(DefineConstants);XPLAT</DefineConstants>
+        <!--NOTE:DevDivPublicKey.snk comes from src\\tools\\devdiv\\FinalPublicKey.snk -->
+        <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)..\Keys\DevDivPublicKey.snk</AssemblyOriginatorKeyFile>
+
+        <!-- Use PublicSign when TEST_LAB_BUILD is enabled -->
+        <PublicSign Condition="'$(TEST_LAB_BUILD)'!=''">true</PublicSign>
+        <DelaySign Condition="'$(TEST_LAB_BUILD)'==''">true</DelaySign>
       </PropertyGroup>
     </When>
     <Otherwise>
       <PropertyGroup>
-        <IsXPlat>false</IsXPlat>
+        <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)..\Keys\ExternalKey.snk</AssemblyOriginatorKeyFile>
+        <DelaySign>false</DelaySign>
       </PropertyGroup>
     </Otherwise>
   </Choose>
-
+  
   <!-- Symbol Settings -->
   <Choose>
     <When Condition="$(Configuration.Contains('Debug'))">
       <PropertyGroup>
         <DebugSymbols>true</DebugSymbols>
-        <DebugType>full</DebugType>
+        <DebugType Condition="$([MSBuild]::IsOSPlatform('windows'))">full</DebugType>
```

Comment:
>DebugType [](http://example.com/codeflow?start=9&length=9)

Can we go further and also check for official builds?

Reply:
Do we only want `full` for official builds but `portable` for everything else?

Reply:
Since this is only debug configurations, I am not sure if we need it at all? But if so: yes, and yes for release configurations.